x/bank/src/keeper.rs

```diff
@@ -485,7 +585,28 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, AK: AuthKeeper<SK, M>, M: Module>
         supply_store.set(
             coin.denom.to_string().into_bytes(),
             coin.amount.to_string().into_bytes(),
-        );
+        )
+    }
+
+    pub fn supply<DB: Database, CTX: TransactionalContext<DB, SK>>(
+        &self,
+        ctx: &mut CTX,
+        denom: &Denom,
+    ) -> Result<Option<Coin>, GasStoreErrors> {
+        let store = ctx.kv_store(&self.store_key);
+        let supply_store = store.prefix_store(SUPPLY_KEY);
+
+        let amount_bytes = supply_store.get(denom.as_ref().as_bytes())?;
+
+        match amount_bytes {
+            Some(bytes) => Ok(Some(Coin {
+                denom: denom.clone(),
+                amount: Uint256::from_str(&String::from_utf8_lossy(&bytes))
```

Comment:
Might be better to not use the lossy conversion here. If it's not valid utf8 the there's a bug or corruption.

Reply:
Keypoint here is not for long, which gives slightly better code